Source/Core/Core/Boot/Boot.cpp

```diff
@@ -351,6 +351,44 @@ bool CBoot::Load_BS2(const std::string& boot_rom_filename)
   // Run the descrambler over the encrypted section containing BS1/BS2
   ExpansionInterface::CEXIIPL::Descrambler((u8*)data.data() + 0x100, 0x1AFE00);
 
+  // Perform a surgical modification on known IPLs for Emulated Memory Size Override.
+  if (Config::Get(Config::MAIN_RAM_OVERRIDE_ENABLE))
+  {
+    // Precalculate the new SIMM value.
+    const s16 mem1_s_size = Common::swap16(Config::Get(Config::MAIN_MEM1_SIZE) >> 16);
+    // [lis	r0, 0x0180] <-- This lis instruction is what sets the Simulated MEM1 Size in retail
+    // IPLs.  We're aiming to replace the SIMM field, which is conveniently a signed short.
+    s16* lis_simm_field;
+    // [ori	r9, r9, 0x0003] <-- this instruction putting the value "3" into r9 causes a stw later
+    // down the line to revert the instruction we're trying to change.  I don't understand why, but
+    // I do know NOPing this away doesn't break anything!
+    u32* ori_that_ruins_things;
+
+    switch (ipl_hash)
+    {
+    case NTSC_v1_0:
+    case NTSC_v1_1:
+    case PAL_v1_0:
+    case MPAL_v1_1:
+      lis_simm_field = (s16*)(data.data() + 0xCB6);         // 0x820 + 0x494 + 2
+      ori_that_ruins_things = (u32*)(data.data() + 0x5FC);  // 0x100 + 0x4fc
```

Comment:
Dereferencing any of these will result in undefined behavior, as this violates C++ strict aliasing rules. If you want to make these modifications, then you need to make use of `std::memcpy` to do it in a well-defined manner e.g.

```cpp
const u32 nop = Common::swap16(0x60000000);
std::memcpy(data.data() + 0x5FC, &nop, sizeof(nop));
std::memcpy(data.data() + 0xCB6, &mem1_size, sizeof(mem1_size));
```